include/device_queue.h

```diff
@@ -40,6 +40,29 @@ namespace detail {
 	}
 #endif
 
+	struct device_allocation {
```

Comment:
I think having this as a move-only type makes sense semantically but given that we don't actually move allocations right now, and this will presumably be replaced by the instruction graph anyway, I've turned it into a POD!